zeroex/orderwatch/order_watcher_test.go

```diff
@@ -1027,6 +1027,279 @@ func TestOrderWatcherCleanup(t *testing.T) {
 	}
 }
 
+// Scenario 1: Header 1 exists in DB. Get's removed and then re-added.
+func TestOrderWatcherUpdateBlockHeadersStoredInDBScenario1(t *testing.T) {
```

Comment:
@fabioberger these tests are isolated, but they are repetitive and IMO overly verbose. Snippets like this are duplicated many times:

```go
miniHeaders := []*miniheader.MiniHeader{}
err = meshDB.MiniHeaders.FindAll(&miniHeaders)
require.NoError(t, err)
assert.Len(t, miniHeaders, 1)
```

It distracts from what we are really trying to test: "this input should result in that output".

What I was suggesting was using table-driven tests, similar to what we do [in the `db` package](https://github.com/0xProject/0x-mesh/blob/dee8353e2e51862920745f5006e5fd38ab162b4b/db/query_test.go#L265) and a few other places in the repo. This style of testing is based on https://github.com/golang/go/wiki/TableDrivenTests and is considered a best practice in the Go ecosystem.

Can you rewrite these tests in the table-driven style? Something like:

```go
testCases := []struct{
    events []*blockwatch.Event
    expectedMiniHeaders []*miniheader.MiniHeader
}{
    {
        events: []*blockwatch.Event{
            // Put the events here. These will be passed directly to updateBlockHeadersStoredInDB
        },
        expectedMiniHeaders: []*miniheader.MiniHeader{
            // Put the expected mini headers here. What do we expect to find in the database after calling updateBlockHeadersStoredInDB?
        }
    }
}

for _, testCase := range testCases {
    // 1. Call updateBlockHeadersStoredInDB
    // 2. Call meshDB.MiniHeaders.FindAll(&miniHeaders)
    // 3. Expect that the miniheaders in the database match testCase.expectedMiniHeaders
}
```

This approach makes the tests far more DRY and easier to understand.

Reply:
I see, sorry for missing that. Done.